src/panels/history/ha-panel-history.ts

```diff
@@ -413,18 +339,20 @@ class HaPanelHistory extends SubscribeMixin(LitElement) {
     if (searchingDeviceId) {
       searchingDeviceId = ensureArray(searchingDeviceId);
       for (const singleSearchingDeviceId of searchingDeviceId) {
-        const foundEntities = this._deviceIdToEntities[singleSearchingDeviceId];
-        if (foundEntities?.length) {
-          for (const foundEntity of foundEntities) {
-            if (foundEntity.entity_category === null) {
-              entityIds.add(foundEntity.entity_id);
-            }
+        const foundEntities = this._deviceEntityLookup[singleSearchingDeviceId];
+        if (!foundEntities?.length) {
+          continue;
+        }
+
+        for (const foundEntity of foundEntities) {
+          if (foundEntity.entity_category === null) {
+            entityIds.add(foundEntity.entity_id);
```

Comment:
This line can cause a separate issue I saw, if you hit return after typing an invalid entity_id the target picker will accept that value. Ideally we should check that this entity_id actually exists, and unfortunately that means you may want to reintroduce the _stateEntities state value to be sure that you're adding a valid entity_id.

Reply:
We should fix this in the picker

AllowCustomValue should be used to allow or not allow

Reply:
Sorry, actually that's line #430 that'g not correct.